### PR TITLE
 Python-gnutls fixed + various new python libraries  

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

python-gnutls needed to discover path of gnutls libs for correct functioning.
Added various new python libraries that are needed for the open source conference server sylkserver (sylkserver.org), which I will submit as a separate pull request. 

Each library is a separate commit to make review easy.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ x ] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by groxxda** [comment]
Please review the indentation (at least lines 1223 1224 11731 11745 11746 17463 20722 look misindented to me), nitpick: missing whitespace in 21242 in front of `]`

**Comment by leenaars** [comment]
@groxxda: fixed all the indents.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1208,6 +1208,21 @@ in modules // {
     };
   });
 
+  attrs = buildPythonPackage {
+    name = "attrs-15.2.0";
+    src = pkgs.fetchurl {
+      url =
+      "https://pypi.python.org/packages/8b/76/c57eefda827b981135ccacd4328fceaa3693f79d9da1e5d78fbe59ebd0c4/attrs-15.2.0.tar.gz";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: can you use the pypi mirror with the old url scheme here and also below. It should work (but occasionally doesn't).

**Comment by leenaars** [comment]
Hi @FRidh,

I was unsure what the best option would be. The old URL scheme is not available anymore.. I really have no clue how to obtain those old urls for new packages. 

Has there been any decision on whether to choose for the CDN (https://pypi.python.org/mirrors) or continue using actual independent mirrors on different URLs (https://www.pypi-mirrors.org) hosted by different organisations - which seems the most robust to me. Currently there is only one mirror listed: 

  # Python PyPI mirrors
  pypi = [
    https://pypi.io/packages/source/
  ];

**Comment by FRidh** [comment]
@leenaars interesting, I wasn't aware of https://pypi-mirrors.org.
While the urls on the websites don't show the old scheme anymore, a translator was added so they should still work. I suggest you have a look at the scheme that is used with all other packages, it's quite easy. As I mentioned the old scheme generally works fine.

**Comment by leenaars** [comment]
Hi @FRidh, changed everything to the old scheme.

**Comment by leenaars** [comment]
BTW: the ordering of packages in python-package.nix is not very clean, to put it mildly.

**Comment by FRidh** [comment]
yep, it gets messier and messier. It would indeed be nice to sorted. If you know of a way... :-)

**Comment by FRidh** [comment]
tests of `attrs` fail with Python 3.5. They need pytest. Likely you also need to invoke pytest. Please test all packages with all interpreters.

Also, since it's going to need some more work, can you also prepend the commit names with `pythonPackages.`.

**Comment by leenaars** [comment]
Hi @FRidh, fixed all isues, rebased, tested. Should be ready.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -1208,6 +1208,26 @@ in modules // {
...
+  attrs = buildPythonPackage (rec {
+    pname = "attrs";
+    version = "16.0.0";
+    name = "attrs-16.0.0";
+    src = pkgs.fetchFromGitHub {
+      owner = "hynek";
+      repo = pname;
+      rev = version;
+      sha256 = "1s28qxv48dhh92vnpnnwgbijv4gcd2vrcnn3g3jm8ag9m8vjhgln";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: PyPI source
https://pypi.python.org/pypi/attrs/16.0.0

**Comment by FRidh** [comment]
`attrs` has tests but they're not being run. If you look in the test folder you can see `pytest` is needed and needs to be invoked.

**Comment by leenaars** [comment]
Hi @FRidh, I had to package Pympler, as the tests were written with that framework. That was broken due to some test that failed. Now builds after patching that test out (which was only relevant on Python 2.5 according to the devs). 

Now builds, added -> huge amount of tests for attrs work perfectly.

**Comment by leenaars** [comment]
Hi @FRidh, is there anything I still need to do?

**Comment by FRidh** [comment]
Yep, `attrs` tests are still not running. And there is two commits that need to be squashed.

**Comment by leenaars** [comment]
Build failure was due to python-gnutls not being available for py3k:
https://github.com/AGProjects/python-gnutls/issues/2

Now it works (in python2.7).

**Comment by leenaars** [comment]
@FRidh : finally got everything working. Should be ready to merge.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -1208,6 +1208,27 @@ in modules // {
     };
   });
 
+  attrs = buildPythonPackage (rec {
+    pname = "attrs";
+    version = "16.0.0";
+    name = "attrs-16.0.0";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why not build it up the name from the arguments that you already gave?

**Pull request merged**